### PR TITLE
Property inspector styling fixes

### LINKED ISSUE #13987: Property Inspector headings use inconsistent text and border styling
## Description

In the left and right panels, headers typically use small text in all caps, with bold weight. For example, here's the Command Palette as it appears when docked to the left: 

image

… and the extension manager, which has expandable sections:

image

The Property Inspector uses larger text, in normal weight, with Title Case for headings:

image

When expanded, the "Common Tools" section has no space between the bottommost control and a boundary line:

image

## Reproduce

1. Open JupyterLab
2. Open the Property Inspector
3. Expand "Common Tools"

## Expected behavior

Headers would be styled like in the extension manager or docked command palette

## Context

- Operating System and version: macOS 12.6.3
- Browser and version: Firefox 102.7.0esr
- JupyterLab version: 4.0.0a33

### PR DESCRIPTION
**Pull request by JasonWeill**
## References

Fixes #13987.

## Code changes

Modifies the property inspector (notebook tools) to use similar styling to the Running Terminals and Kernels UI.

## User-facing changes

The Property Inspector used larger text, in normal weight, with Title Case for headings:

image

When expanded, the "Common Tools" section had no space between the bottommost control and a boundary line:

image

After changing this, the headers use small text in bold:

image

And, when expanded, no stray line appears at the bottom:

image

## Backwards-incompatible changes

None.

### CODE REVIEW

File: packages/ui-components/src/components/collapser.ts [hunk 1]
```diff
@@ -4,7 +4,9 @@
 import { Message } from '@lumino/messaging';
 import { ISignal, Signal } from '@lumino/signaling';
 import { Panel, PanelLayout, Title, Widget } from '@lumino/widgets';
-import { caretDownIcon, caretUpIcon } from '../icon';
+import { caretDownIcon } from '../icon';
+
+const collapsedClass = 'jp-Collapse-header-collapsed';
```
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:9]: Should we use all caps for consts to be consistent with other places. For example:
https://github.com/jupyterlab/jupyterlab/blob/3ecf75ac8eda5f11e55f318658577bba227ecdf6/packages/cells/src/collapser.tsx#L16

```suggestion
const COLLAPSED_CLASS = 'jp-Collapse-header-collapsed';
```

File: packages/ui-components/src/components/collapser.ts [hunk 2]
```diff
@@ -18,6 +20,17 @@ export class Collapser<T extends Widget = Widget> extends Widget {
     this.addClass('jp-Collapse');
```
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:20]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

File: packages/ui-components/src/components/collapser.ts [hunk 3]
```diff
@@ -18,6 +20,17 @@ export class Collapser<T extends Widget = Widget> extends Widget {
...
     this._header = new Widget();
     this._header.addClass('jp-Collapse-header');
```
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:22]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

File: packages/ui-components/src/components/collapser.ts [hunk 4]
```diff
@@ -18,6 +20,17 @@ export class Collapser<T extends Widget = Widget> extends Widget {
...
+    if (collapsed) {
+      this._header.addClass(collapsedClass);
+    }
+    this._header.node.appendChild(caretDownIcon.element({
+      className: 'jp-Collapser-icon'
```
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:27]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

File: packages/ui-components/src/components/collapser.ts [hunk 5]
```diff
@@ -18,6 +20,17 @@ export class Collapser<T extends Widget = Widget> extends Widget {
...
+    }));
+    const titleSpan = document.createElement('span');
+    titleSpan.classList.add('jp-Collapser-title');
```
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:30]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

File: packages/ui-components/src/components/collapser.ts [hunk 6]
```diff
@@ -18,6 +20,17 @@ export class Collapser<T extends Widget = Widget> extends Widget {
...
+    titleSpan.textContent = widget.title.label;
+    this._header.node.appendChild(titleSpan);
+
     this._content = new Panel();
     this._content.addClass('jp-Collapse-contents');
```
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:35]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

File: packages/ui-components/src/components/collapser.ts (hunk 2 above)
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:20]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

File: packages/ui-components/src/components/collapser.ts (hunk 3 above)
Comment (andrii-i) [inline packages/ui-components/src/components/collapser.ts:22]: Should we change this to use all caps global const instead? See https://github.com/jupyterlab/jupyterlab/pull/14069/files#r1115181206

**Review by andrii-i — COMMENTED** [review]
Fixes the problem, line at the bottom of common tools section in property inspector now has spacing vs next section. Please see some suggestions on displaying consts.

**Comment by krassowski** [comment]
bot please update galata snapshots

**Comment by krassowski** [comment]
bot please update documentation snapshots

**Review by andrii-i — APPROVED** [review]

**Comment by krassowski** [comment]
Kicking CI

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @JasonWeill 

It seems the new structure broke the integrationt tests for the metadataform feature and a snapshot was wrongly updated: galata/test/documentation/debugger.test.ts-snapshots/debugger-activate-documentation-linux.png

**Comment by JasonWeill** [comment]
bot please update galata snapshots

**Comment by JasonWeill** [comment]
please update galata snapshots

**Comment by JasonWeill** [comment]
bot please update documentation snapshots

**Comment by JasonWeill** [comment]
@krassowski @fcollonval I'd like to update the UI test screen shots to reflect the new visual style; I reviewed the test reports, and the revised screen shots are correct. I've been asking the bot to update screen shots, but nothing seems to be getting updated. Is there something else that I can do to move this PR forward? Thanks!

**Comment by krassowski** [comment]
Can you try first merging the master branch in? Is there anything specific in the action logs?

**Comment by krassowski** [comment]
This one sorts to be still running: https://github.com/jupyterlab/jupyterlab/actions/runs/4308717001

**Comment by JasonWeill** [comment]
bot please update documentation snapshots

**Comment by JasonWeill** [comment]
@fcollonval After rebasing this branch and regenerating snapshots successfully, CI is now green on this PR ✅

**Comment by krassowski** [comment]
@JasonWeill the CI suite has not run fully yet because bot does not have permissions to trigger them. Kicking CI to run them now.

**Comment by fcollonval** [comment]
bot please update documentation snapshots

**Comment by fcollonval** [comment]
Kicking the CI

File: galata/test/jupyterlab/metadataform.test.ts [hunk 7]
```diff
@@ -76,7 +76,7 @@ async function openForm(
   await activatePropertyInspector(page);
 
   const form = page.locator(
-    `.jp-NotebookTools .jp-Collapse:has(>.jp-Collapse-header:text('${label}'))`
+    `.jp-NotebookTools .jp-Collapse .jp-Collapse-header:hasText('${label}')`
```
Comment (krassowski) [inline galata/test/jupyterlab/metadataform.test.ts:79]:
```suggestion
    '.jp-NotebookTools .jp-Collapse .jp-Collapse-header',
    {hasText: label}
```

File: galata/test/jupyterlab/metadataform.test.ts [hunk 8]
```diff
@@ -160,7 +160,7 @@ test.describe('Required metadata', () => {
 
     // Retrieves the form from its header's text, it should be collapsed.
     const form = page.locator(
-      '.jp-NotebookTools .jp-Collapse:has(>.jp-Collapse-header:text("Extension metadata"))'
+      `.jp-NotebookTools .jp-Collapse .jp-Collapse-header:hasText('Extension metadata')`
```
Comment (krassowski) [inline galata/test/jupyterlab/metadataform.test.ts:163]:
```suggestion
      '.jp-NotebookTools .jp-Collapse .jp-Collapse-header',
      {hasText: 'Extension metadata'}
```

**Review by krassowski — COMMENTED** [review]
Could we use locator `options`?

**Comment by fcollonval** [comment]
bot please update documentation snapshots

**Comment by JasonWeill** [comment]
bot please update documentation snapshots

**Comment by JasonWeill** [comment]
please update galata snapshots

File: packages/ui-components/style/collapse.css [hunk 9]
```diff
@@ -7,19 +7,32 @@
...
-  font-size: var(--jp-ui-font-size2);
+  border-bottom: solid var(--jp-border-width) var(--jp-border-color2);
+  color: var(--jp-ui-font-color1);
+  cursor: pointer;
+  display: flex;
+  align-items: center;
+  font-size: var(--jp-ui-font-size0);
+  font-weight: 600;
+  text-transform: uppercase;
+}
```
Comment (krassowski) [inline packages/ui-components/style/collapse.css:23]:
```suggestion
  user-select: none;
}
```

One difference I noted when testing on binder is that when clicking on headers in property inspector the titles sometimes get selected whereas in extension manager or debugger this does not happen thanks to `user-select: none`.

**Review by krassowski — APPROVED** [review]
Thank you @JasonWeill. I left one minor suggestion to further harmonise behaviour/improve UX.

**Comment by JasonWeill** [comment]
bot please update documentation snapshots

**Comment by JasonWeill** [comment]
@jtpio @fcollonval I'd like to test these changes with my team using a new alpha build. It's been about 9 days since 4.0.0a35 was released. What's the process to get 4.0.0a36 released from tip of `master`? Thanks!

**Comment by jtpio** [comment]
@JasonWeill we use the Jupyter Releaser for making releases: https://github.com/jupyterlab/jupyterlab/blob/master/RELEASE.md#the-jupyterlab-release-process

There is no requirement of time between two releases, so there could be a new one yes.
Would you like to give it a try? You would need to configure the releaser tokens (more details in the releaser docs) and have maintainer access to `jupyterlab` on PyPI and `npm`. Happy to discuss on other mediums like Gitter to help set you up.

**Comment by fcollonval** [comment]
> If you can make it today, it will be great - I was waiting for https://github.com/jupyterlab/jupyterlab/pull/13875 that is now ready to be merged. But require a release right after to get the CI green. So if you could merge it and release, it will be awesome.

**Comment by jtpio** [comment]
FYI I just submitted https://github.com/jupyterlab/jupyterlab/pull/14176 to help make JupyterLab releases from the repo directly and to avoid setting up the tokens and the fork.

**Pull request merged**